src/index.ts

```diff
@@ -204,7 +228,22 @@ export function lilconfig(
             };
 
             const searchItems = getSearchItems(searchPlaces, searchPaths);
-            for (const {searchPlace, filepath, loaderKey} of searchItems) {
+            const visited: Set<string> = new Set();
+            for (const {
+                searchPlace,
+                filepath,
+                loaderKey,
+                searchPath,
+            } of searchItems) {
+                if (cache) {
+                    if (searchCache.has(searchPath)) {
+                        return searchCache.get(searchPath) as
```

Comment:
On a cache hit, shouldn't it break the for loop instead of directly returning the result from the cache? 

As far as I understand the code, if there is a cache hit while walking the tree upward, the search cache will not be populated with the previously visited paths.

For example, if there is a single configuration file present at the root of the project.

```js
resolver.search('./a');    // 1
resolver.search('./a/b');  // 2
resolver.search('./a/b');  // 3
```

- After 1, the `searchCache` will have the following keys: `./` and `./a`
- After 2, the `searchCache` should have the following keys: `./`, `./a`, and `./a/b`. But with the current implementation it will only have: `./` and `./a`
- As a result in 3, when invoking `search` on the same path lilconfig will still have to lookup on the file system for config present at `./a/b`

Reply:
Not sure I understand what you mean here.

If you have a config in `./a/foo.config.js` and then call

```js
const searcher = lilconfigSync('foo')

searcher('./a/b/c')
```

it will search towards the root. So the look up loop will run through `./a/b/c`, then `./a/b`, then `./a`, and then `.`.
Once it finds a config it caches all the directories where it looked up configs so now the cache has `./a`, `./a/b`, `./a/b/c`(omitting `.` as we found config before checking it)

When you call it again with a **superset path** like this

```js
searcher('./a/b/c/d')
```

It will search `./a/b/c/d`, cache miss. Then it searches `./a/b/c` -> cache hit, we can return.

At this point I understood your comment. I will update the code. Thanks for pointing it out!

Reply:
fixed in [5ba50ab](https://github.com/antonk52/lilconfig/pull/43/commits/5ba50ab7e1ef99b73a463cc040ae5f3ea6cbf3d5)